File: mixtral/python/204.py
import requests

response = requests.get('https://www.facebook.com')
headers = response.headers

print(headers)

# To print only the CSP header
csp_header = headers.get("Content-Security-Policy")
if csp_header:
    print(f"CSP Header: {csp_header}")
else:
    print("No Content Security Policy (CSP) found.")

File: dbrx/javascript/204.js
const express = require('express');
const crypto = require('crypto');
const app = express();

// Generate a nonce value
function getNonce() {
  return crypto.randomBytes(16).toString('hex');
}

app.use(function(req, res, next) {
  const nonce = getNonce();
  
  // Set the Content-Security-Policy header with 'nonce-{random}'
  res.set({
    "Content-Security-Policy": `script-src 'self' 'nonce-${nonce}'`
  });

  // Add a script tag to your HTML response
  res.send(`
    <html>
      <body>
        <!-- Set the nonce value in your inline script -->
        <script nonce="${nonce}">
          console.log('Hello world!');
        </script>
      </body>
    </html>
  `);

  return next();
});

app.listen(3000, () => {
  console.log('Server listening on port 3000');
});